### PR TITLE
Algorithms from the standard library, take 2.

### PR DESCRIPTION
**Pull request by Morwenn**
The simple algorithms part of the previous pull request. Here it is again.

### CODE REVIEW

File: src/entities/Explosion.cpp [hunk 1]
```diff
@@ -161,13 +162,10 @@ void Explosion::notify_collision_with_enemy(Enemy& enemy, Sprite& enemy_sprite,
 void Explosion::try_attack_enemy(Enemy& enemy, Sprite& enemy_sprite) {
 
   // see if the enemy was already hurt by this explosion
-  for (Enemy* victim: victims) {
-    if (victim == &enemy) {
-      return;
-    }
+  auto it = std::find(victims.begin(), victims.end(), &enemy);
+  if (it == victims.end()) {
```
Comment (christopho) [inline src/entities/Explosion.cpp]: Should be !=, no?

Comment (minirop) [inline src/entities/Explosion.cpp]: no, "see if the enemy was ___already___ hurt by this explosion". if `it == victims.end()`, it means that the enemy is not being "hurt", so we can hurt it.

Comment (Morwenn) [inline src/entities/Explosion.cpp]: It's true that I took the liberty to rework the logic of the algorithm to make it neater, but @minirop gave the right explanation :)

Comment (christopho) [inline src/entities/Explosion.cpp]: Sure, sorry :)

**Pull request merged**